lib/spack/spack/solver/concretize.lp

```diff
@@ -464,7 +464,9 @@ version_weight(Package, Weight) :- external_version(Package, Version, Weight).
 version(Package, Version) :- external_version(Package, Version, Weight).
 
 % if a package is not buildable (external_only), only externals are allowed
-external(Package) :- external_only(Package), node(Package).
+external(Package) :- external_only(Package),
```

Comment:
I still think we shouldn't be lying to Spack about whether it's an external. If it's a previously installed external, we should be marking that appropriately, not pretending it's not an external.

Reply:
Imo the visible effect to users is consistent. The property is called `buildable` (not `external_only`), and the user can require constraints to be sure clingo selects a spec matching the external.